### PR TITLE
adding rootless podman functionality

### PR DESCRIPTION
**Pull request by tbelway**
Compatibility with rootless podman added.

### CODE REVIEW

File: gravity-sync [hunk 1]
```diff
@@ -1465,8 +1482,14 @@ function validate_ph_folders {
         PH_FTL_CHECK=$(sudo podman container ls | grep "${PIHOLE_CONTAINER_IMAGE}")
         if [ "$PH_FTL_CHECK" == "" ]; then
             MESSAGE="${UI_VALIDATING_FAIL_CONTAINER} ${UI_CORE_APP}"
-            echo_fail
-            exit_no_change
+            echo "failed bitch"
```
Comment (vmstan) [inline gravity-sync:1485]: ?

Comment (tbelway) [reply gravity-sync:1485]: Omg lol that was not supposed to be there. I'll fix it. 
Was a test for something being dumb...

**Comment by tbelway** [comment]
I have it working with no errors atm. Let me know if you need anything else.

**Comment by vmstan** [comment]
Thanks for this, I need to get some time to properly test this before merging.

**Comment by tbelway** [comment]
I have it working with no errors atm. Let me know if you need anything else.

> Thanks for this, I need to get some time to properly test this before merging.

No worries bud. I am also hella sleep deprived with my kids, so just let me know if there's anything that doesn't play nice with your tests and I'll correct. I have it working for me, it plays seamless with fedora server and almalinux but for some reason I sometimes get gravity.db corruptions on microos, but I had that before I was using gravity sync so I am pretty sure it has to do with something else :)

I had it syncing across 3 pihole instances in a loop fashion, which is not resilient in the case of one going down, so if I ever get some free time maybe I'll look at a spoke model...

**Comment by RhavinX** [comment]
Any updates on getting this tested/merged? I run rootless podman and would really like this functionality.
As it stands on the machine running rootless podman:

✓ Podman installation has been detected
! There is no Podman container of Pi-hole running
✗ Pi-hole was not found on this system
» Status Report
✗ 1 critical issue(s) prevent successful deployment
```

CONTAINER ID  IMAGE                            COMMAND      CREATED             STATUS                       PORTS                                                                                                       NAMES
aa74381baf23  docker.io/mvance/unbound:latest  /unbound.sh  About a minute ago  Up About a minute (healthy)  0.0.0.0:53->53/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:5335->5335/tcp, 0.0.0.0:53->53/udp, 0.0.0.0:5335->5335/udp  unbound
12c193d96ca1  docker.io/pihole/pihole:latest                About a minute ago  Up About a minute (healthy)  0.0.0.0:53->53/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:5335->5335/tcp, 0.0.0.0:53->53/udp, 0.0.0.0:5335->5335/udp  pihole
```

**Comment by tbelway** [comment]
> Any updates on getting this tested/merged? I run rootless podman and would really like this functionality. As it stands on the machine running rootless podman:
> 
> ✓ Podman installation has been detected ! There is no Podman container of Pi-hole running ✗ Pi-hole was not found on this system » Status Report ✗ 1 critical issue(s) prevent successful deployment
> 
> ```
> 
> 
> CONTAINER ID  IMAGE                            COMMAND      CREATED         STATUS                   PORTS                                                                                                       NAMES
> aa74381baf23  docker.io/mvance/unbound:latest  /unbound.sh  21 minutes ago  Up 21 minutes (healthy)  0.0.0.0:53->53/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:5335->5335/tcp, 0.0.0.0:53->53/udp, 0.0.0.0:5335->5335/udp  unbound
> 12c193d96ca1  docker.io/pihole/pihole:latest                21 minutes ago  Up 21 minutes (healthy)  0.0.0.0:53->53/tcp, 0.0.0.0:80->80/tcp, 0.0.0.0:5335->5335/tcp, 0.0.0.0:53->53/udp, 0.0.0.0:5335->5335/udp  pihole
> ```

I'm using it across three piholes at home, two of which are rootless podman one of which is a standard VM.
I've tested across:
fedora/almalinux VM rootless podman (I've done both)
microOS rootless podman (does have an odd bug where sometimes the gravity.db corrupts, but I'm pretty sure I've narrowed down the root cause. This happens regardless of if I'm using gravity-sync or not.
Ubuntu standard install.

Feel free to test it out for yourself and see if you have issues :) You could always help out vmstan by reporting your feedback too.

**Comment by vmstan** [comment]
Effective July 26, 2024, this project has been retired. Thank you for your use and enthusiasm for a project that began as a few lines of bash in a Slack channel and envolved into something far more complex, and used by many thousands of Pi-hole enthusiasts over the last four years.

**Pull request closed without merging**